app/assets/javascripts/dashboard/views/dash/explorative_annotations_view.js

```diff
@@ -321,12 +406,28 @@ export default class ExplorativeAnnotationsView extends React.PureComponent {
     );
   }
 
+  renderSearchTags() {
+    return this.state.tags.map(tag =>
+      <Tag
+        key={tag}
+        color={TemplateHelpers.stringToColor(tag)}
+        afterClose={_.partial(this.removeTagFromSearch, tag)}
+        closable
+      >
+        {tag}
+      </Tag>,
+    );
+  }
+
   render() {
     const marginRight = { marginRight: 8 };
     const search = (
       <Search
         style={{ width: 200, float: "right" }}
-        onPressEnter={this.handleSearch}
+        onPressEnter={(event: SyntheticInputEvent) => {
+          this.addTagToSearch(event.target.value);
+          this.setState({ searchQuery: "" });
```

Comment:
This should clear the search box too, which is why `searchQuery` should be passed to `<Search />`.